core/src/main/java/org/apache/iceberg/ReachableFileUtil.java

```diff
@@ -103,14 +106,24 @@ private static TableMetadata findFirstExistentPreviousMetadata(
    * @return the location of manifest Lists
    */
   public static List<String> manifestListLocations(Table table) {
-    Iterable<Snapshot> snapshots = table.snapshots();
-    List<String> manifestListLocations = Lists.newArrayList();
-    for (Snapshot snapshot : snapshots) {
-      String manifestListLocation = snapshot.manifestListLocation();
-      if (manifestListLocation != null) {
-        manifestListLocations.add(manifestListLocation);
-      }
+    return manifestListLocations(table, null);
+  }
+
+  /**
+   * Returns locations of manifest lists in a table.
+   *
+   * @param table table for which manifestList needs to be fetched
+   * @param snapshots ids of snapshots for which manifest lists will be returned
+   * @return the location of manifest Lists
+   */
+  public static List<String> manifestListLocations(Table table, Set<Long> snapshots) {
+    Stream<Snapshot> snapshotStream = StreamSupport.stream(table.snapshots().spliterator(), false);
```

Comment:
Minor: rather than using Java's `spliterator` method and streams, we generally prefer to use `Iterables.filter` and `Iterables.transform` or a simple `for` loop.

Reply:
Done, used Iterables.filter and restored the old for loop